api/gateway-operator/v2alpha1/controlplane_types.go

```diff
@@ -228,6 +235,25 @@ type ControlPlaneK8sCache struct {
 	InitSyncDuration *metav1.Duration `json:"initSyncDuration,omitempty"`
 }
 
+// ControlPlaneConfigDump defines the options for dumping translated Kong configuration from a diagnostics server.
+//
+// +apireference:kgo:include
+// +kubebuilder:validation:XValidation:message="Cannot enable dumpSensitive when enabled is set to false",rule="self.enabled || !self.dumpSensitive"
+type ControlPlaneConfigDump struct {
+	// When Enabled is true, Operator will dump the translated Kong configuration by it from a diagnostics server.
+	//
+	// +required
+	// +kubebuilder:default=false
+	Enabled bool `json:"enabled"`
```

Comment:
We'll want to introduce `nobools` linter at some point (I've already pushed https://github.com/Kong/kubernetes-configuration/pull/526 for this).

So I'd suggest to change this e.g. `state: enabled|disable`. WDYT?

Reply:
Done.